JavaScript/FastNoiseLite.js

```diff
@@ -0,0 +1,3372 @@
+		 * @param {number} y
+		 * @param {number} z
+		 * @returns {number}
+		 */
+		let R3 = (x, y, z) => {
+			let seed = this.#Seed;
+			let sum = 0;
+			let amp = this.#FractalBounding;
+
+			for (let i = 0; i < this.#Octaves; i++) {
+				let noise = Math.abs(this.#GenNoiseSingle(seed++, x, y, z));
+				sum += (noise * -2 + 1) * amp;
+				amp *= FastNoise.#Lerp(1.0, 1 - noise, this.#WeightedStrength);
+
+				x *= this.#Lacunarity;
+				y *= this.#Lacunarity;
+				z *= this.#Lacunarity;
+				amp *= this.#Gain;
+			}
+			return sum;
+		};
+
+		if (arguments.length === 2) {
+			return R2(arguments[0], arguments[1]);
+		}
+
+		if (arguments.length === 3) {
+			return R3(arguments[0], arguments[1], arguments[2]);
+		}
+	}
+
+	/**
+	 * @private
+	 * @returns {number}
+	 */
+	#GenFractalPingPong() {
+		/**
+		 *
+		 * @param {number} x
+		 * @param {number} y
+		 * @returns {number}
+		 */
+		let R2 = (x, y) => {
+			let seed = this.#Seed;
+			let sum = 0;
+			let amp = this.#FractalBounding;
+
+			for (let i = 0; i < this.#Octaves; i++) {
+				let noise = FastNoise.#PingPong(
+					(this.#GenNoiseSingle(seed++, x, y) + 1) * this.#PingPongStrength
+				);
+				sum += (noise - 0.5) * 2 * amp;
+				amp *= FastNoise.#Lerp(1.0, noise, this.#WeightedStrength);
+
+				x *= this.#Lacunarity;
+				y *= this.#Lacunarity;
+				amp *= this.#Gain;
+			}
+			return sum;
+		};
+
+		/**
+		 *
+		 * @param {number} x
+		 * @param {number} y
+		 * @param {number} z
+		 * @returns {number}
+		 */
+		let R3 = (x, y, z) => {
+			let seed = this.#Seed;
+			let sum = 0;
+			let amp = this.#FractalBounding;
+
+			for (let i = 0; i < this.#Octaves; i++) {
+				let noise = FastNoise.#PingPong(
+					(this.#GenNoiseSingle(seed++, x, y, z) + 1) * this.#PingPongStrength
+				);
+				sum += (noise - 0.5) * 2 * amp;
+				amp *= FastNoise.#Lerp(1.0, noise, this.#WeightedStrength);
+
+				x *= this.#Lacunarity;
+				y *= this.#Lacunarity;
+				z *= this.#Lacunarity;
+				amp *= this.#Gain;
+			}
+			return sum;
+		};
+		if (arguments.length === 2) {
+			return R2(arguments[0], arguments[1]);
+		}
+
+		if (arguments.length === 3) {
+			return R3(arguments[0], arguments[1], arguments[2]);
+		}
+	}
+
+	/**
+	 * @private
+	 * @returns {number}
+	 */
+	#SingleOpenSimplex2() {
+		/**
+		 *
+		 * @param {number} seed
+		 * @param {number} x
+		 * @param {number} y
+		 * @returns {number}
+		 */
+		let R2 = (seed, x, y) => {
+			const SQRT3 = 1.7320508075688772935274463415059;
+			const G2 = (3 - SQRT3) / 6;
+
+			let i = Math.floor(x);
+			let j = Math.floor(y);
+			let xi = x - i;
+			let yi = y - j;
+
+			let t = (xi + yi) * G2;
+			let x0 = xi - t;
+			let y0 = yi - t;
+
+			i *= this.#PrimeX;
+			j *= this.#PrimeY;
+
+			let n0, n1, n2;
+
+			let a = 0.5 - x0 * x0 - y0 * y0;
+
+			if (a <= 0) {
+				n0 = 0;
+			} else {
+				n0 = a * a * (a * a) * this.#GradCoord(seed, i, j, x0, y0);
+			}
+
+			let c = 2 * (1 - 2 * G2) * (1 / G2 - 2) * t + (-2 * (1 - 2 * G2) * (1 - 2 * G2) + a);
```

Comment:
If it is determined that method-level `const` declarations are not pre-computed during method parsing, and incurring a speed penalty, then perhaps it would also be worth it to move the various constant expression parts like these out into their own class-level constants. If you don't know, though, I also probably wouldn't worry about it too much.